src/main/java/frc/team2412/robot/Robot.java

```diff
@@ -230,29 +232,60 @@ public boolean isCompetition() {
 	private boolean wasArmButtonPressed = false;
 	private boolean isArmCoast = false;
 
+	private Optional<Boolean> wasAlignmentCorrect = Optional.empty();
+
 	@Override
 	public void disabledInit() {
 		Shuffleboard.stopRecording();
-		wasArmButtonPressed = subsystems.armSubsystem.isIdleModeTogglePressed();
+		if (subsystems.armSubsystem != null) {
+			wasArmButtonPressed = subsystems.armSubsystem.isIdleModeTogglePressed();
+		}
+		wasAlignmentCorrect = Optional.empty();
 	}
 
 	@Override
 	public void disabledPeriodic() {
-		boolean isArmButtonPressed = subsystems.armSubsystem.isIdleModeTogglePressed();
-		if (!wasArmButtonPressed && isArmButtonPressed) {
-			if (isArmCoast) {
-				subsystems.armSubsystem.setBrake();
-			} else {
-				subsystems.armSubsystem.setCoast();
+		if (subsystems.armSubsystem != null) {
+			boolean isArmButtonPressed = subsystems.armSubsystem.isIdleModeTogglePressed();
+			if (!wasArmButtonPressed && isArmButtonPressed) {
+				if (isArmCoast) {
+					subsystems.armSubsystem.setBrake();
+				} else {
+					subsystems.armSubsystem.setCoast();
+				}
+				isArmCoast = !isArmCoast;
+			}
+			wasArmButtonPressed = isArmButtonPressed;
+		}
+
+		if ((subsystems.visionSubsystem != null) && (subsystems.armLedSubsystem != null)) {
+			// spotless:off
+			// Check if there is a recent target, and if the angle is close to 0
+			boolean isAlignmentCorrect =
+					subsystems.visionSubsystem.hasTargets()
+					&& ((subsystems.visionSubsystem.getLastTimestampSeconds() - Timer.getFPGATimestamp())
+						< 0.5)
```

Comment:
Can delete this, as it's always negative since the order of arguments is wrong & the `hasTargets()` above is immediately false if no targets are visible.